File: frontend/src/lib/stores/status.ts
import { writable, derived } from 'svelte/store'
import { selectedAccountId } from './accounts'

export interface AccountStatus {
  status: number
  color: string
}

// Map of accountId -> status
export const statuses = writable<Record<number, AccountStatus>>({})

export const selectedStatus = derived(
  [statuses, selectedAccountId],
  ([$statuses, $id]) => $id != null ? $statuses[$id] ?? { status: 0, color: 'black' } : { status: 0, color: 'black' }
)

export const statusLabels: Record<number, string> = {
  0: 'Offline',
  1: 'Starting',
  2: 'Online',
  3: 'Pausing',
  4: 'Paused',
  5: 'Stopping'
}

File: internal/domain/enum/status.go
package enum

type Status int

const (
	StatusOffline  Status = 0
	StatusStarting Status = 1
	StatusOnline   Status = 2
	StatusPausing  Status = 3
	StatusPaused   Status = 4
	StatusStopping Status = 5
)

func (s Status) String() string {
	switch s {
	case StatusOffline:
		return "Offline"
	case StatusStarting:
		return "Starting"
	case StatusOnline:
		return "Online"
	case StatusPausing:
		return "Pausing"
	case StatusPaused:
		return "Paused"
	case StatusStopping:
		return "Stopping"
	default:
		return "Unknown"
	}
}

// Color returns a CSS color string for UI display.
func (s Status) Color() string {
	switch s {
	case StatusOnline:
		return "green"
	case StatusStarting, StatusPausing, StatusStopping:
		return "orange"
	case StatusPaused:
		return "red"
	default:
		return "black"
	}
}
